specs/protocol/holocene/derivation.md

```diff
@@ -0,0 +1,186 @@
+# Holocene Chain Derivation Changes
+
+# Stricter Derivation Rules
+
+The Holocene hardfork introduces several changes to batch derivation rules that render the
+derivation pipeline stricter and simpler.
+The changes are:
+
+- _Steady Batch Derivation_ derives invalid singluar batches immediately as deposit-only
+blocks.
+- _Partial Span Batch Validity_ determines the validity of singular batches within a span batch
+individually, only invalidating the remaining span batch upon the first invalid batch.
+- _Strict Batch Ordering_ required batches within and across channels to be strictly ordered.
+
+The channel and frame format are not changed, see also [A note on the current channel
+format](#a-note-on-the-current-channel-format) for a discussion about a possible future
+simplification of the channel and frame format under the stricter derivation rules.
+
+## Steady Batch Derivation
+
+Steady Batch Derivation changes the derivation rules for invalid batches, and instead of giving them
+a second chance for a worst-case full sequencing window, they are immediately derived as
+deposit-only blocks. There are two stages at which this new rule takes effect, in the batch queue
+and in the engine queue.
+
+### Batch Queue
+
+With Holocene, whereever a singular batch would be dropped before, it is now immediately replaced
+by a deposit-only batch.
+
+TODO: details
+
+### Engine Queue
+
+Holocene changes the way in which the Engine Queue responds to payload attributes errors: On a
+payload processing error, the payload is immediately replaced by the respecive empty deposit-only
+attributes for the same block number and payload processing is reattempted with these new
+attributes. As before, a failure to process deposit-only attributes is a critical error.
+
+This new rule applies to both, processing of attributes to advance the safe or unsafe chain.
+This is in contract to just dropping the payload pre-Holocene and waiting for possible future valid
+payload attributes.
+
+## Partial Span Batch Validity
+
+Partial Span Batch Validity changes the span batch validation rules by only invalidating the
+remaining span batch once an invalid singular batch is encountered. Prior to this, span batch
+validity was treated atomically, so earlier pending valid singular batches could be invalidated by a
+later invalid singular batch pulled from a span batch.
+To accomplish this new behavior, the [Delta span batch batch queue rules](../delta/span-batches.md#batch-queue)
+are changed in the following ways.
+
+The max sequencer time drift check is applied to each singluar batch derived from a span batch
+individually. Once a singular batch is found to be invalid, only the remaining span batch is
+dropped. Note that the sequencing window check still applies to the span batch in full, because the
+oldest blocks would be the first to be invalidated and if dropped would also invalidate newer
+blocks.
```

Comment:
Does this mean that a singular batch that was marked valid or turned into a deposit-only batch could still be invalidated due to the sequencing window check applying to the span batch in full? I don't really understand the "span batch in full" wording here.

Reply:
Spec got reworked. New rules drop invalid batches, only replacing invalid payloads with deposit-only blocks.